subsys/mgmt/mcumgr/transport/src/smp_udp.c

```diff
@@ -22,26 +22,40 @@
 #include <zephyr/mgmt/mcumgr/smp/smp.h>
 #include <zephyr/mgmt/mcumgr/transport/smp.h>
 #include <zephyr/mgmt/mcumgr/transport/smp_udp.h>
-#include <errno.h>
-
-#include <mgmt/mcumgr/transport/smp_internal.h>
-
-#ifdef CONFIG_MCUMGR_TRANSPORT_UDP_AUTOMATIC_INIT
 #include <zephyr/mgmt/mcumgr/mgmt/handlers.h>
 #include <zephyr/net/net_mgmt.h>
 #include <zephyr/net/net_event.h>
 #include <zephyr/net/conn_mgr.h>
-#endif
+#include <errno.h>
+
+#include <mgmt/mcumgr/transport/smp_internal.h>
 
 #define LOG_LEVEL CONFIG_MCUMGR_LOG_LEVEL
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(smp_udp);
 
 BUILD_ASSERT(CONFIG_MCUMGR_TRANSPORT_UDP_MTU != 0, "CONFIG_MCUMGR_TRANSPORT_UDP_MTU must be > 0");
 
+#if !defined(CONFIG_MCUMGR_TRANSPORT_UDP_IPV4) && !defined(CONFIG_MCUMGR_TRANSPORT_UDP_IPV6)
+BUILD_ASSERT(0, "Either IPv4 or IPv6 SMP must be enabled for the MCUmgr UDP SMP transport using "
+		"CONFIG_MCUMGR_TRANSPORT_UDP_IPV4 or CONFIG_MCUMGR_TRANSPORT_UDP_IPV6");
+#endif
+
+#define IS_THREAD_RUNNING(thread)					\
+	((thread.base.thread_state == _THREAD_PENDING ||		\
+	  thread.base.thread_state == _THREAD_PRESTART ||		\
+	  thread.base.thread_state == _THREAD_SUSPENDED ||		\
+	  thread.base.thread_state == _THREAD_QUEUED) ? true : false)
```

Comment:
I am surprised to find out that there is no API for retrieving the thread status.
What about applying a single bitwise statement?
```suggestion
#define IS_THREAD_RUNNING(thread)                                       \
	((thread.base.thread_state & (_THREAD_PENDING |                 \
				      _THREAD_PRESTART |                \
				      _THREAD_SUSPENDED |               \
				      _THREAD_QUEUED)) ? true : false)
```

Reply:
Fixed